pkg/security/podsecuritypolicy/selinux/mustrunas.go

```diff
@@ -74,3 +81,128 @@ func (s *mustRunAs) Validate(fldPath *field.Path, _ *api.Pod, _ *api.Container,
 
 	return allErrs
 }
+
+func equalLevels(expectedLevel, actualLevel string) (bool, error) {
+	if expectedLevel == actualLevel {
+		return true, nil
+	}
+
+	// Level format: https://selinuxproject.org/page/MLSStatements#level
+
+	// "s0:c6,c0" => [ "s0", "c6,c0" ]
```

Comment:
All this parsing seems super brittle to me, and I'd really prefer that it was done with a regular expression...



Not going to require it, but maybe at least a TODO?

Reply:
Ok, I've added a TODO.

Reply:
In future it should be possible to use `CanonicalizeContext()` from selinux: https://github.com/opencontainers/selinux/pull/19 Here is also a related discussion: https://github.com/openshift/origin/pull/16432#issuecomment-338716218